src/Domain/FunctionsOfTime/PiecewisePolynomial.cpp

```diff
@@ -141,6 +154,31 @@ void PiecewisePolynomial<MaxDeriv>::update(const double time_of_update,
 
   func[MaxDeriv] = std::move(updated_max_deriv);
   store_entry(time_of_update, std::move(func), next_expiration_time);
+
+  // This update time came from the backlog. Remove it and return so we don't
+  // keep recursing.
+  if (update_backlog_.contains(time_of_update)) {
+    update_backlog_.erase(time_of_update);
+    return;
+  }
+
+  // We will only get here in the "highest level" of the recursion
+  for (auto it = update_backlog_.begin(); it != update_backlog_.end();) {
```

Comment:
would a simple for loop be easier to read, rather than recursion?

```
for (auto it = update_backlog_.begin(); it != update_backlog_.end();) {
    auto next_it = std::next(it);
    if (it->first == time_bounds()[1]) {
      store_entry(it->first, std::move(it->second.first), it->second.second);
      update_backlog_.erase(time_of_update);
    }
    it = next_it;
}
```
This way `update` can do all checks and then either (add it to the backlog and return) or (store entry and apply the backlog which has already been checked).